### PR TITLE
[Merged by Bors] - feat(LinearAlgebra/Matrix/Adjugate): adjugate_fin_three, adjugate_fin_three_of

### PR DESCRIPTION
**Pull request by Bergschaf**
Add `adjugate_fin_three` and `adjugate_fin_three_of` theorems, in the same style as `adjugate_fin_two`.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/LinearAlgebra/Matrix/Adjugate.lean
Comment (eric-wieser) [inline Mathlib/LinearAlgebra/Matrix/Adjugate.lean:431]: Most of the `()`s are not needed here (they're not in the `fin_two` version either); I also suggest tidying the indentation a little to highlight where each row begins:
```suggestion
    !![
      A 1 1 * A 2 2 - A 1 2 * A 2 1,
        -(A 0 1 * A 2 2) + A 0 2 * A 2 1,
        A 0 1 * A 1 2 - A 0 2 * A 1 1;
      -(A 1 0 * A 2 2) + A 1 2 * A 2 0,
        A 0 0 * A 2 2 - A 0 2 * A 2 0,
        -(A 0 0 * A 1 2) + A 0 2 * A 1 0;
      A 1 0 * A 2 1 - A 1 1 * A 2 0,
        -(A 0 0 * A 2 1) + A 0 1 * A 2 0,
        A 0 0 * A 1 1 - A 0 1 * A 1 0] := by
```

File: Mathlib/LinearAlgebra/Matrix/Adjugate.lean
Comment (eric-wieser) [inline Mathlib/LinearAlgebra/Matrix/Adjugate.lean:420]: I don't think this is needed, since it's exactly the statement of the second lemma anyway.
```suggestion

```

File: Mathlib/LinearAlgebra/Matrix/Adjugate.lean
Comment (eric-wieser) [inline Mathlib/LinearAlgebra/Matrix/Adjugate.lean:623]:
```suggestion
```

`#align`s are only for code that existed in mathlib 3

File: Mathlib/LinearAlgebra/Matrix/Adjugate.lean
Comment (eric-wieser) [inline Mathlib/LinearAlgebra/Matrix/Adjugate.lean:434]: Whenever you  use a tactic like `fin_cases` that creates multiple goals, you should mark each goal with a bullet point, as
```lean
fin_cases i
· sorry
· sorry
· sorry
```
where the three `sorry`s will be pieces of your proof below.

**Review by eric-wieser — COMMENTED** [review]
Thanks! I agree we should have these results, though I think we should try to get the proof a (lot!) shorter first. In the meantime, I have a few style suggestions.

File: Mathlib/LinearAlgebra/Matrix/Adjugate.lean
Comment (Bergschaf) [reply Mathlib/LinearAlgebra/Matrix/Adjugate.lean:434]: I will work on this later (when school is over lol). How would you shorten the proof?

**Comment by tb65536** [comment]
If you move this result to after `adjugate_fin_succ_eq_det_submatrix`, then I think you can give a three-line proof:
```
  ext i j
  rw [adjugate_fin_succ_eq_det_submatrix, det_fin_two]
  fin_cases i <;> fin_cases j <;> simp [updateRow, Fin.succAbove, Fin.lt_def] <;> ring
```

File: Mathlib/LinearAlgebra/Matrix/Adjugate.lean
Comment (Ruben-VandeVelde) [inline Mathlib/LinearAlgebra/Matrix/Adjugate.lean:450]:
```suggestion
  adjugate_fin_three _
```

**Review by Ruben-VandeVelde — APPROVED** [review]
Stylistic suggestions; seems fine otherwise

File: Mathlib/LinearAlgebra/Matrix/Adjugate.lean
Comment (eric-wieser) [inline Mathlib/LinearAlgebra/Matrix/Adjugate.lean:449]: Perhaps with some column alignment:
```suggestion
    adjugate !![a, b, c; d, e, f; g, h, i] =
      !![  e * i  - f * h, -(b * i) + c * h,   b * f  - c * e;
         -(d * i) + f * g,   a * i  - c * g, -(a * f) + c * d;
           d * h  - e * g, -(a * h) + b * g,   a * e  - b * d] :=
```

File: Mathlib/LinearAlgebra/Matrix/Adjugate.lean
Comment (tb65536) [inline Mathlib/LinearAlgebra/Matrix/Adjugate.lean:441]:
```suggestion
  ext i j
  rw [adjugate_fin_succ_eq_det_submatrix, det_fin_two]
  fin_cases i <;> fin_cases j <;> simp [updateRow, Fin.succAbove, Fin.lt_def] <;> ring
```

**Comment by tb65536** [comment]
Is it worth taking the time to golf `adjugate_fin_two` while we're at it?

**Comment by Bergschaf** [comment]
I can loop into it and create a separate PR if i find something

**Comment by Bergschaf** [comment]
#11051

**Comment by tb65536** [comment]
You'll need to merge master once #11051 is merged.

File: Mathlib/LinearAlgebra/Matrix/Adjugate.lean
Comment (eric-wieser) [inline Mathlib/LinearAlgebra/Matrix/Adjugate.lean:438]:
```suggestion
theorem adjugate_fin_three_of (a b c d e f g h i : α) :
```

File: Mathlib/LinearAlgebra/Matrix/Adjugate.lean
Comment (eric-wieser) [inline Mathlib/LinearAlgebra/Matrix/Adjugate.lean:421]: Just noting there's a stray blank line here:
```suggestion
```

**Review by eric-wieser — APPROVED** [review]
bors d+

Thanks!

**Comment by tb65536** [comment]
You should usually wait for green CI before merging, just to be on the safe side.

**Pull request closed without merging**
